encfs/autosprintf.cpp

```diff
@@ -2,20 +2,18 @@
    Copyright (C) 2002 Free Software Foundation, Inc.
    Written by Bruno Haible <bruno@clisp.org>, 2002.
 
-   This program is free software; you can redistribute it and/or modify it
-   under the terms of the GNU Library General Public License as published
-   by the Free Software Foundation; either version 2, or (at your option)
-   any later version.
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Library General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
```

Comment:
Note: this comment does not block merging.

This phrasing is already in here (I didn't notice it before).  Something to consider going forward: https://opensource.stackexchange.com/questions/1777/why-is-gplv2-incompatible-with-gplv3

Quote from commenter "DevSolar":

> GPLv2+ code that's been "upped" to GPLv3 has to remain GPLv3. The person you got the GPLv2+ code from has no way to merge back any downstream (GPLv3) changes back into his (GPLv2+) codebase.

Should we consider removing `or (at your option) any later version` or accept if anybody just so happens to want to merge GPLv3 code back to us we decide later to upgrade this project to GPLv3?

In general, I'm not personally comfortable with open ended legal comments like that.

Reply:
It is normal ended and the same phrase was in original license notice.

Reply:
I don't think it's normal nor acceptable to have such open ended terms.  My personal opinion of course.  I would be OK if GPLv3 was compatible with GPLv2 but that's not how it is in the currently language of GPLv3.  "Or later version" is super open ended legally and could mean anything down the line.  I respectfully disagree with your opinion.

Reply:
Not sure what you mean with "not normal", this is the standard text from GNU https://www.gnu.org/licenses/gpl-2.0.html#SEC4

Reply:
In legal terms it's not normal in my opinion.